lib/hls/hls_parser.js

```diff
@@ -120,6 +121,16 @@ shaka.hls.HlsParser = class {
      */
     this.updatePlaylistDelay_ = 0;
 
+    /**
+     * A time offset to apply to EXT-X-PROGRAM-DATE-TIME values to normalize
+     * them so that they start at 0. This is necessary because the streaming
+     * engine assumes these values are in the timescale of the presentation.
```

Comment:
A small nit: timescale has a specific meaning as a unit, and this could be confusing.  How about:

> This is necessary because these times will be used to set presentation times for segments.  `null` means we don't have any data.

Reply:
Done.